### ISSUE TITLE
Performance during MiTM

### ISSUE THREAD

**Issue by hazcod** [telanflow/mps#6]
Hi, just to let you know that when using `mitmHandler`, a lot of CPU time is spent in `mitm_handler.transmit`.

![image](https://user-images.githubusercontent.com/5222512/97708064-77bb4b80-1ab8-11eb-8c3a-fd038477a4a2.png)

Not sure if the client-side connections are pooled?

**Comment by telanflow** [comment]
Yes, I know. MITM does not use connection pooling, and I am still considering a solution.

The tunnel agent has used the connection pool.

https://github.com/telanflow/mps/blob/668b4ac974bbc66848672bcdd0c2150a5b74406c/mitm_handler.go#L133-L193

**Comment by telanflow** [comment]
the reason for the `TLS handshake` should be that you did not reuse the connection for each request.

You need to pay attention to:

The client needs to reuse the connection every time it initiates a http request. Otherwise the http request will re-create the connection, resulting in repeated `TLS handshake`

**Comment by hazcod** [comment]
Hi @telanflow , i'll check the http protocol version.
So what does this todo refer to then? https://github.com/telanflow/mps/blob/master/mitm_handler.go#L134

**Comment by hazcod** [comment]
@telanflow Interesting, I've disabled `connection: close` on request & response and always use `HTTP/1.1`.
![image](https://user-images.githubusercontent.com/5222512/97806462-04523f00-1c5c-11eb-9917-6349c230ba64.png)
